docs/source/tutorial_tipsntricks.rst

```diff
@@ -44,6 +44,39 @@ That is, by adding a colon ":" followed by the flag name, an equals sign, and th
 You can add multiple flags, all separated by ":".
 
 
+Self-Chats
+##########
+
+Sometimes it is useful to generate models talking to themselves. You can do this with:
+
+.. code-block:: python
+
+   # Self-chatting Poly-Encoder model on ConvAI2
+   python parlai/scripts/self_chat.py -mf zoo:pretrained_transformers/model_poly/model -t convai2:selfchat --inference topk -ne 10 --display-examples True -dt valid
+
+The particular task requires a world for the agents that handles the particular nature of the task, 
+see e.g. `here <https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/convai2/worlds.py#L98>`_ 
+or `here <https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/wizard_of_wikipedia/worlds.py#L106>`_.
+If the model does not need to run on a particular task you can also use:
+  
+
+.. code-block:: python
+
+   # Self-chatting Poly-Encoder model on a generic task (so e.g., no ConvAI2 personas are input)
+   python parlai/scripts/self_chat.py -mf zoo:pretrained_transformers/model_poly/model -t self_chat --inference topk -ne 10 --display-examples True -dt valid
+
+
+Prettifying Display of Chats
+############################
+
+This handy script can prettify the display of json file of chats (sequences of parlai messages):
```

Comment:
does the data need to be in a certain format to be prettified?

Reply:
it does but i think we need to fix that to be more permissive of any json of parlai messages, so i didn't really want to document the current given that it should be fixed. what i do give are example files so they can be looked at..

Reply:
works for me